src/services/refactors/extractSymbol.ts

```diff
@@ -1547,13 +1548,13 @@ function getContainingVariableDeclarationIfInList(node: Node, scope: Scope) {
     }
 }
 
-function getFirstDeclaration(type: Type): Declaration | undefined {
+function getFirstDeclarationBeforePosition(type: Type, position: number): Declaration | undefined {

```

Comment:
What exactly was the problem? What is the thing that was captured before a specific position?

Reply:
The problem was that type variables declared inside the extraction range where used in the type parameter list of the new function, which isn't needed because the types will be declared in the function's body instead. So with this change I'm only including the type parameters in scope that are declared before the extraction's range.

As for the respective [crash](https://github.com/microsoft/tsserverfuzzer/issues/433), I'm not too sure. I debugged this a bit with @sandersn a few weeks ago and we suspect that there's an issue with the printer inserting trivia nodes in the incorrect places.

Reply:
*Okay*, so this is a really just a proxy/hack for saying "get me the free (captured) type variables" - because any captured/outer type parameter must occur strictly before the function itself.

Reply:
Exactly.